src/lib/ppx_dhall_type/deriving.ml

```diff
@@ -0,0 +1,122 @@
+(* deriving.ml -- deriving ppx for Dhall types *)
+
+(* TODO:
+    deriver for signatures
+    default values in records
+*)
+
+open Core_kernel
+open Ppxlib
+
+let deriver = "dhall_type"
+
+let make_lident_cmp items lident =
+  List.mem items (Longident.name lident.txt) ~equal:String.equal
+
+let is_bool_lident = make_lident_cmp ["bool"; "Bool.t"]
+
+let is_int_lident = make_lident_cmp ["int"; "Int.t"]
+
+let is_float_lident = make_lident_cmp ["float"; "Float.t"]
+
+let is_string_lident = make_lident_cmp ["string"; "String.t"]
+
+let is_option_lident = make_lident_cmp ["option"; "Option.t"]
+
+let is_list_lident = make_lident_cmp ["list"; "List.t"]
+
+let rec dhall_type_of_core_type core_type =
+  let (module Ast_builder) = Ast_builder.make core_type.ptyp_loc in
+  let open Ast_builder in
+  match core_type.ptyp_desc with
+  | Ptyp_constr (lident, []) when is_bool_lident lident ->
+      [%expr Bool]
```

Comment:
Nit: it would be better to fully qualify these instead of using an `open`, so that there is no shadowing.